xpk.py

```diff
@@ -2673,6 +2698,222 @@ def workload_list(args) -> None:
   xpk_exit(0)
 
 
+def inspector_run_command_helper(args, command, command_description, file) -> int:
+  """Runs a command for xpk inspector, and build the output file.
+
+  Args:
+    args: user provided arguments for running the command.
+    command: the cli command to run.
+    command_description: a brief description of the command run.
+    file: file to add command output to.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  prefix = f'Command: {command}\nCommand Description: {command_description}\n'
+  postfix = '========================================================'
+  return_code, command_output = run_command_for_value(command, f'{command_description}', args)
+
+  if return_code != 0:
+    xpk_print(f'{command} returned ERROR {return_code} with output: {command_output}')
+    return 1
+
+  inspector_command_output = f'{prefix} \n{command_output} \n{postfix} \n'
+  append_temporary_file(inspector_command_output, file)
+
+  if args.print_to_terminal:
+    xpk_print(inspector_command_output)
+  return 0
+
+
+def inspector_run_workload_list_helper(args, command_description, file) -> int:
+  """Runs a workload list command for xpk inspector, and build the output file.
+
+  Args:
+    args: user provided arguments for running the command.
+    command_description: a brief description of the command run.
+    file: file to add command output to.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  prefix = f'Command Description: {command_description}\n'
+  postfix = '========================================================'
+  return_code, command_output = get_workload_list(args)
+  if return_code != 0:
+    xpk_exit(return_code)
+  inspector_command_output = f'{prefix} \n{command_output} \n{postfix} \n'
+  append_temporary_file(inspector_command_output, file)
+  if args.print_to_terminal:
+    xpk_print(inspector_command_output)
+  return 0
+
+
+def inspector_output_link_helper(args, link, link_description, file) -> int:
+  """Outputs a link for xpk inspector to the output file.
+
+  Args:
+    args: user provided arguments for.
+    link: link to output.
+    link_description: describes what the link is for.
+    file: file to add command output to.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  inspector_link = (
+    f'Link Description: {link_description}\n'
+    f'Link: {link}\n'
+    '========================================================'
+  )
+  append_temporary_file(inspector_link, file)
+  if args.print_to_terminal:
+    xpk_print(inspector_link)
+  return 0
+
+
+def inspector(args) -> int:
+  """Function around inspector which investigates failures in the kueue.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  # Future Improvements for inspector:
+  # 1. Print out the resource flavor.
+  # 2. List what is next in Queue.
+  # 3. Split inspector into different subcommands to parse info easier.
```

Comment:
Is there a reason why we cannot split inspector right now to get info for all workloads and specific workload separately?

Reply:
Good thoughts! How should we split inspector separately? 

It could be capacity inspection / workload inspection / kueue and jobset inspection. Basically I don't see clear dimensions to split across yet.

Reply:
We can look at splitting this later when we have a better view!

Reply:
Sounds good!